newIDE/app/src/ProjectManager/index.js

```diff
@@ -165,6 +176,61 @@ class Item extends React.Component<ItemProps, {||}> {
       <div style={styles.itemName}>{this.props.primaryText}</div>
     );
 
+    const menuTemplate = [
+      {
+        label: 'Edit',
+        click: () => this.props.onEdit(),
+      },
+      {
+        label: 'Rename',
+        click: () => this.props.onEditName(),
+      },
+      {
+        label: 'Delete',
+        click: () => this.props.onDelete(),
+      },
+      {
+        label: this.props.addLabel,
+        visible: !!this.props.onAdd,
+        click: () => this.props.onAdd(),
+      },
+      { type: 'separator' },
+      {
+        label: 'Copy',
+        click: () => this.props.onCopy(),
+      },
+      {
+        label: 'Cut',
+        click: () => this.props.onCut(),
+      },
+      {
+        label: 'Paste',
+        enabled: this.props.canPaste(),
```

Comment:
This line is a problem. Can you see why? 

The idea of having buildMenuTemplate being a function is that you:
* Don't pay for the construction of the menu template when it's not needed (here, you create one for each item... for large projects with a layout of scenes/etc this can start to be noticeable).
* You only construct the menu at the time it's needed... so you can inspect things like the clipboard to enable/disable menu items.

Here `canPaste` is called at the time the item is rendered... but what if the clipboard was changed since then? :)

Solution: move this back to the buildMenuTemplate function

Reply:
Ohh okay, got it.. sorry, didn't think about that :sweat_smile:.
I'll fix this right away.

Reply:
Done. I've just pulled out `buildMenuTemplate` into a class function to make the render function a bit cleaner. I hope it's fine now.

Reply:
Can we actually keep it in the render? I know it's counter intuitive when used to React classes, but as an effort to migrate progressively components to React functions with hooks, which are shorter and safer, it's actually ok to keep stuff in render (unless it's not good for readability or performance). Here I think it's totally fair to consider the menu items as part of the render - it's the user interface :)

Reply:
Oh okay right, done.